lib/CL/pocl_util.c

```diff
@@ -2478,42 +2513,6 @@ pocl_update_event_finished (cl_int status, const char *func, unsigned line,
   POCL_UNLOCK_OBJ (event);
   ops->broadcast (event);
 
-  /* With remote being asynchronous it is possible that an event is signaled as
```

Comment:
if an event is marked as failed, it will lock (when broadcasting), notify and mark all events in it's notify list as failed as well. event chains that look something like the image will be deadlocked
![event_chain](https://github.com/pocl/pocl/assets/15857200/47146bc3-d6f7-4663-ac9c-edcc40d5078f)

Reply:
i've put this here as a comment since it is about code that is being removed. I could add a comment above the broadcast function warning about the locking behavior it has.

Reply:
I think adding more code documentation would be beneficial, yes.

Reply:
i've left a warning in the pocl_cl.h description of the broadcast function